### PR TITLE
This resolves issue #9492 removing the escapeHtml function from core.…

### PR DESCRIPTION
**Pull request by Jordan-Rob**
…js and placing it in utils/text.js

_Please check the following:_

-   [X] Do the tests still pass?[^1]
-   [X] Does the code comply with the style guide?
    -   [X] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [X] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [X] **Please list the exact browser and operating system versions you tested**:
          Chrome on MacOS Monterey

    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.
Tests should be carried out normally no special conditions are required

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

**Comment by Jordan-Rob** [comment]
Hi @lb- Here's the PR I opened to sort out issue #9492 although a test seems to be failing yet they are passing locally

Screen Shot 2022-10-31 at 1 08 42 PM

File: client/src/entrypoints/admin/core.js [hunk 1]
```diff
@@ -15,19 +15,6 @@ function addMessage(status, text) {
...
-    '<': '&lt;',
-    '>': '&gt;',
-    '"': '&quot;',
-    "'": '&#039;',
-  };
-
-  return text.replace(/[&<>"']/g, (char) => map[char]);
-}
-
-window.escapeHtml = escapeHtml;
```
Comment (gasman) [inline client/src/entrypoints/admin/core.js:30]: Removing this `window.escapeHtml = escapeHtml;` line will mean that the function is no longer available globally to all Javascript - only to files that explicitly import it from utils. Right now it looks like add-multiple.js [in wagtailimages](https://github.com/wagtail/wagtail/blob/05951b350196b059307d4c0233fdf773d4b41486/wagtail/images/static_src/wagtailimages/js/add-multiple.js#L38) and [wagtaildocs](https://github.com/wagtail/wagtail/blob/05951b350196b059307d4c0233fdf773d4b41486/wagtail/documents/static_src/wagtaildocs/js/add-multiple.js#L27) relies on the global definition.

We should either have core.js import the definition of `escapeHtml` from utils/text.js, and then make it available globally by keeping this line in place - or, preferably, move the two add-multiple.js files into the appropriate places in client/src/entrypoints and update them to explicitly import `escapeHtml` from utils. (That might be out of scope for this issue, though.)

**Review by gasman — CHANGES_REQUESTED** [review]

File: client/src/entrypoints/admin/core.js (hunk 1 above)
Comment (Jordan-Rob) [reply client/src/entrypoints/admin/core.js:30]: okay noted thanks let me make the changes

**Comment by lb-** [comment]
@Jordan-Rob the failure is due to linting (code formatting). If you run `npm run format` - it will fix up any formatting issues.

File: client/src/entrypoints/admin/core.js (hunk 1 above)
Comment (lb-) [reply client/src/entrypoints/admin/core.js:30]: Yes. As per the issue checklist - we need to ensure the existing code can still access the global. 

Or, as per above suggestion - we can import the util via moving the other files that use it. Although that may be a harder - but feel free to give it a go.

**Review by gasman — APPROVED** [review]

**Review by lb- — APPROVED** [review]
Great, this works well locally - thanks @Jordan-Rob 
Screen Shot 2022-11-02 at 6 35 18 am

**Comment by lb-** [comment]
@Jordan-Rob  thanks so much for this, and epic work digging into a solid solution here where actually had two definitions of the same function.

I have added your name to the contributors list as 'Jordan Rob', let me know if you want a different name listed.

**Comment by Jordan-Rob** [comment]
> @Jordan-Rob thanks so much for this, and epic work digging into a solid solution here where actually had two definitions of the same function.
> 
> I have added your name to the contributors list as 'Jordan Rob', let me know if you want a different name listed.

You're welcome, glad to have been of help. Yeah 'Jordan Rob is fine'

**Pull request merged**